packages/accounts/src/msca/plugins/multisig/extension.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  AccountNotFoundError,
+  type GetAccountParameter,
+  type IsUndefined,
+  type SmartContractAccount,
+} from "@alchemy/aa-core";
+import { type Address, type Chain, type Client, type Transport } from "viem";
+import type { GetPluginAddressParameter } from "../types.js";
+import {
+  MultisigPlugin,
+  multisigPluginActions as multisigPluginActions_,
+  type MultisigPluginActions as MultisigPluginActions_,
+} from "./plugin.js";
+
+export type MultisigPluginActions<
+  TAccount extends SmartContractAccount | undefined =
+    | SmartContractAccount
+    | undefined
+> = MultisigPluginActions_<TAccount> & {
+  readOwners: (
+    params: GetPluginAddressParameter & GetAccountParameter<TAccount>
+  ) => Promise<ReadonlyArray<Address>>;
+
+  isOwnerOf: (
+    params: { address: Address } & GetPluginAddressParameter &
+      GetAccountParameter<TAccount>
+  ) => Promise<boolean>;
+
+  getThreshold: (
+    params: GetPluginAddressParameter & GetAccountParameter<TAccount>
+  ) => Promise<bigint>;
+} & (IsUndefined<TAccount> extends false
+    ? {
+        readOwners: (
+          params?: GetPluginAddressParameter & GetAccountParameter<TAccount>
+        ) => Promise<ReadonlyArray<Address>>;
+      }
+    : {});
+
+export const multisigPluginActions: <
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TAccount extends SmartContractAccount | undefined =
+    | SmartContractAccount
+    | undefined
+>(
+  client: Client<TTransport, TChain, TAccount>
+) => MultisigPluginActions<TAccount> = <
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TAccount extends SmartContractAccount | undefined =
+    | SmartContractAccount
+    | undefined
+>(
+  client: Client<TTransport, TChain, TAccount>
+) => ({
+  ...multisigPluginActions_(client),
+  async readOwners(
+    args: GetPluginAddressParameter & GetAccountParameter<TAccount>
+  ) {
+    const account = args?.account ?? client.account;
+    if (!account) {
+      throw new AccountNotFoundError();
+    }
+    // TODO: check if the account actually has the plugin installed
+    // either via account loupe or checking if the supports interface call passes on the account
```

Comment:
comment just as a reminder

learn: why would we ever do option 2 over option 1? 1 seems canonical to the 6900 ecosystem.

Reply:
this logic is repeated across these methods. consider a private helper method to share?

Reply:
Good question - my guess is that loupe is optional for ERC6900 accounts so option 2 would be for non-MSCA/MA accounts that are ERC6900 compliant

Reply:
oh i see, the loupe is an MA thing?

Reply:
yup